xoa/data_source_vms.go

```diff
@@ -0,0 +1,107 @@
+package xoa
+
+import (
+	"log"
+	"strings"
+
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceXoaVms() *schema.Resource {
+
+	return &schema.Resource{
+		Read: dataSourceVmsRead,
+		Schema: map[string]*schema.Schema{
+			"vms": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     resourceVm(),
+			},
+			"num": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"pool_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"container": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"power_state": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```

Comment:
Another way we could design this is to take the aws amis data source approach ([docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami_ids)). Since the underlying XO api allows us to provide any key, value pairs we want, we could allow a user to filter by any field they want even if we didn't identify their use case ahead of time.

I know the hosts data source took a different direction, but I figured I would mention this. I'm ok with implementing this either way, but since this idea just occurred to me I figured I would mention it.

Reply:
Yea, I think we are far enough along on this solution that we should stick to the current direction. But something to think about if we need to continue to add more support for additional filtering or for future data sources.

Reply:
Created a new issue to keep track of that.